internal/provider/connection_resource.go

```diff
@@ -69,28 +71,22 @@ func (r *ConnectionResource) Schema(ctx context.Context, req resource.SchemaRequ
 				},
 				Optional: true,
 				Attributes: map[string]schema.Attribute{
-					"streams": schema.ListNestedAttribute{
+					"streams": schema.SetNestedAttribute{
 						Computed: true,
-						PlanModifiers: []planmodifier.List{
-							speakeasy_listplanmodifier.SuppressDiff(speakeasy_listplanmodifier.ExplicitSuppress),
-						},
 						Optional: true,
+						PlanModifiers: []planmodifier.Set{
+							speakeasy_setplanmodifier.UniqueByKey("name"),
+						},
 						NestedObject: schema.NestedAttributeObject{
 							Attributes: map[string]schema.Attribute{
 								"cursor_field": schema.ListAttribute{
-									Computed: true,
-									PlanModifiers: []planmodifier.List{
```

Comment:
I'm not entirely certain that we need to remove these plan modifiers, so I defer to someone with more background to correct me here if this is not the right approach